packages/neon-core/src/u/misc.ts

```diff
@@ -78,3 +79,44 @@ export function reverseHex(hex: string): string {
   }
   return out;
 }
+
+interface NeonSerializable {
+  size: number;
+  serialize: () => string;
+}
+type Serializables = number | HexString | NeonSerializable[];
+/**
+ * Calculates the byte size of any supported input following NEO's variable int format.
+ */
+export function getSerializedSize(value: Serializables): number {
+  switch (typeof value) {
+    case "number": {
+      if (value < 0xfd) return 1;
+      else if (value <= 0xffff) return 3;
+      else return 5;

```

Comment:
I do think my previous suggestion for a 9 byte var int still stands. Not clear why would you think this link is about reading byte array? The byte array method is located above and its called `ReadVarBytes`. This is called `readVarInt`.

https://github.com/neo-project/neo/blob/ab7ec04925c1e10c9338e3deb8880e93b75ff306/src/neo/IO/Helper.cs#L211

Reply:
I should have originally written: "reading **from** bytearrays (==from the binary stream)".

Your `readVarInt` reads **from** a byte array (`BinaryReader` is a stream of bytes) into an `unsigned long`. It does not apply here because it always determines how many bytes to read next from the stream based on the first byte. If we applied the logic linked then
- value < 0xFD = length 1
- value == 0xFD = length 3
- value == 0xFE = length 4
- value == 0xFF = length 8

compare this to
- getSerializedSize(0xFC) = length 1 (OK)
- getSerializedSize(0xFD) = length 3 (OK)
- getSerializedSize(0xFE) = length 3 (Differs)
- getSerializedSize(0xFF) = length 3 (Differs)

We can't compare a serialized format vs a consumed format.
```C#
var num = 0xFFFF;
MemoryStream ms = new MemoryStream();
var writer = new BinaryWriter(ms);
writer.Write(num); // Note how we write a number in regular format not in varInt format
byte[] serialized = ms.ToArray();
Console.WriteLine(serialized.ToHexString()); // FFFF0000

MemoryStream ms2 = new MemoryStream(serialized);
var reader = new BinaryReader(ms2);
Console.WriteLine(Neo.IO.Helper.ReadVarInt(reader)); // exception: Unable to read beyond the end of the stream.
// because it will read the first byte (0xFF) and think it needs to read 8 more to get the integer value while there 
// are only 4 bytes in the stream in total 
```

Reply:
An Witness array of length larger than 0xffffffff will break this code but fortunately, C# is unable to deal with it.